### PR TITLE
prevent default in video player on key down

### PR DESCRIPTION
**Pull request by NomadicDeveloper22**
add `e.preventDefault();` when pressing 'j' 'k' or 'l' while playing video. Prevents Firefox from opening "find in page" when "Search for text when you start typing" is enabled in browser settings

**Fix:** #4908

### CODE REVIEW

**Comment by dmitrylyzo** [comment]
What about `f`, `m`, `p`, `n`, `0..9`?

I think we can add `preventDefault` for any browser (omit `if (browser.firefox)`).

**Comment by dmitrylyzo** [comment]
> Would it make sense to just add a single `preventDefault` for the whole function? Or at least just before the switch itself. Instead of adding a line for each switch case.

Probably, otherwise browser will start searching on other keys. But this at least breaks `Enter`, `F5`, `F11`, `F12`.

> Or would that have a negative impact on certain cases like for gamepads (don't have one to test myself)

Only if gamepad has bindings in the browser or system. :man_shrugging:

**Comment by TheRedSpy15** [comment]
I opted for specifying which cases need the `e.preventDefault()` (letters and nums) to prevent any issue with other keys

I also rearranged the switch cases so the letters and numbers are after everything else. Just to make it a bit easier when looking at

File: src/controllers/playback/video/index.js [hunk 1]
```diff
@@ -1227,50 +1233,22 @@ export default function (view) {
                     e.stopPropagation();
                 }
                 break;
-            case 'k':
-                playbackManager.playPause(currentPlayer);
-                showOsd(btnPlayPause);
-                break;
             case 'ArrowUp':
-            case 'Up':
```
Comment (dmitrylyzo) [inline src/controllers/playback/video/index.js:1235]: I don't think it should be split up.

File: src/controllers/playback/video/index.js [hunk 2]
```diff
@@ -1227,50 +1233,22 @@ export default function (view) {
...
                 playbackManager.volumeUp(currentPlayer);
                 break;
             case 'ArrowDown':
-            case 'Down':
```
Comment (dmitrylyzo) [inline src/controllers/playback/video/index.js:1239]: same

File: src/controllers/playback/video/index.js [hunk 3]
```diff
@@ -1227,50 +1233,22 @@ export default function (view) {
...
                 playbackManager.volumeDown(currentPlayer);
                 break;
-            case 'l':
```
Comment (dmitrylyzo) [inline src/controllers/playback/video/index.js:1242]: IMO, it is better to keep one command handler - less code to maintain. Also, there is no harm in preventing default action for arrow keys.

File: src/controllers/playback/video/index.js [hunk 4]
```diff
@@ -1227,50 +1233,22 @@ export default function (view) {
...
             case 'ArrowRight':
             case 'Right':
                 playbackManager.fastForward(currentPlayer);
                 showOsd(btnFastForward);
                 break;
-            case 'j':
```
Comment (dmitrylyzo) [inline src/controllers/playback/video/index.js:1248]: same

File: src/controllers/playback/video/index.js [hunk 5]
```diff
@@ -1219,6 +1219,12 @@ export default function (view) {
             case 'Enter':
                 showOsd();
                 break;
+            case 'PageUp':
+                playbackManager.nextChapter(currentPlayer);
+                break;
+            case 'PageDown':
+                playbackManager.previousChapter(currentPlayer);
+                break;
```
Comment (dmitrylyzo) [inline src/controllers/playback/video/index.js:1227]: I'm curious. Why did they climb so high?

Comment (TheRedSpy15) [reply src/controllers/playback/video/index.js:1227]: No functional reason, other than it was easier on the eyes to organize by which needed the preventDefault and which didn't

File: src/controllers/playback/video/index.js (hunk 1 above)
Comment (TheRedSpy15) [reply src/controllers/playback/video/index.js:1235]: Ah that's my mistake. I just added a new commit to fix this and the other similar mistakes

**Review by dmitrylyzo — COMMENTED** [review]
Right now, I'm torn between preventing the default action for handled commands only (current implementation) and preventing it for all alphanumeric keys.

**Comment by cpmsmith** [comment]
> Right now, I'm torn between preventing the default action for handled commands only (current implementation) and preventing it for all alphanumeric keys.

I definitely think it's more proper to `preventDefault` on handled commands only. We don't know what users have bound to any given key. In particular since it doesn't always check modifier keys, if we unconditionally `preventDefault` we'd be preventing most common keyboard shortcuts.

**Pull request closed without merging**